packages/metrics/scripts/reportGenerator.ts

```diff
@@ -0,0 +1,859 @@
+          informantDOB: '',
+          informantMaritalStatus: '',
+          informantOccupation: '',
+          informantCity: '',
+          informantDistrict: '',
+          informantState: '',
+          informantRelationship: ''
+        }
+
+        //Address
+        deathRow.healthCenter = fullComposition.healthCenter ?? ''
+        deathRow.officeLocation = fullComposition.officeLocation ?? ''
+        deathRow.deathDistrict = fullComposition.deceased?.district ?? ''
+        deathRow.deathState = fullComposition.deceased?.state ?? ''
+        deathRow.deathCity = fullComposition.deceased?.city ?? ''
+
+        //Deceased details
+        deathRow.deceasedGen = fullComposition.deceased?.gender ?? ''
+        deathRow.deceasedDOB = fullComposition.deceased?.birthDate ?? ''
+        deathRow.deceasedMaritalStatus =
+          fullComposition.deceased?.maritalStatus ?? ''
+        deathRow.deceasedDate = fullComposition.deceased?.deceasedDate ?? ''
+
+        //Informant details
+        deathRow.informantDOB = fullComposition.informant?.birthDate ?? ''
+        deathRow.informantMaritalStatus =
+          fullComposition.informant?.maritalStatus ?? ''
+        deathRow.informantOccupation =
+          fullComposition.informant?.occupation ?? ''
+        deathRow.informantCity = fullComposition.informant?.city ?? ''
+        deathRow.informantDistrict = fullComposition.informant?.district ?? ''
+        deathRow.informantState = fullComposition.informant?.state ?? ''
+        deathRow.informantRelationship =
+          fullComposition.informant?.relationship ?? ''
+
+        //Observations
+        deathRow.uncertifiedMannerOfDeath =
+          fullComposition.observations?.uncertifiedMannerOfDeath ?? ''
+        deathRow.verbalAutopsyDescription =
+          fullComposition.observations?.verbalAutopsyDescription ?? ''
+        deathRow.causeOfDeathMethod =
+          fullComposition.observations?.causeOfDeathMethod ?? ''
+        deathRow.causeOfDeathEstablished = fullComposition.observations
+          ?.causeOfDeathEstablished
+          ? 'Yes'
+          : 'No'
+        deathRow.causeOfDeath = fullComposition.observations?.causeOfDeath ?? ''
+        deathRow.numMaleDependentsOnDeceased =
+          fullComposition.observations?.numMaleDependentsOnDeceased ?? ''
+        deathRow.numFemaleDependentsOnDeceased =
+          fullComposition.observations?.numFemaleDependentsOnDeceased ?? ''
+
+        // tslint:disable-next-line
+        console.log('Writing CSV row for death declaration...')
+        await deathCSVWriter.writeRecords([deathRow])
+      }
+    }
+    // tslint:disable-next-line
+    console.log(
+      'Successfully generated CSV report for birth and death declarations.'
+    )
+  } catch (error) {
+    // tslint:disable-next-line
+    console.log('Sorry. Something went wrong!', error)
+  }
+}
+
+const startScript = async () => {
+  await connect()
+  const compositionsCursor = await getCompositionCursor()
```

Comment:
@sadmananik @Zangetsu101  we need a way to broadcast the progress of this script in a useful way.  The current console logs are not good enough as they provide no information to the user how much longer there is to run and they will get lost in all the other logs from the metrics microservice.  I think that we should refactor this file a bit a bit and split the start date and end date of the year into months and use some kind of common log message format  so we can use Kibana tags to filter them...  

1. First, create the csv file so that it is ready to be appended to ...
2. Split the submitted year dates into months.
3. For each month 

a) Log one line:  "VSEXPORT GENERATION: Commencing vital statistics export for month 1 of 12"
b) Get all the composition ids and locations
c) Create a calculation to be able to log percentage progress like this:   "VSEXPORT GENERATION: Vital statistics export for month 1 at: 0%"

My concern is that in production, this script may take a long time and it will be hard to understand why the server may be slow.

Reply:
I have modified the script.